challenges/advanced-typegaurd/question.py

```diff
@@ -0,0 +1,22 @@
+"""
+TODO:
+
+is_string is a function that returns a boolean result. If the result is True,
```

Comment:
I feel we can be more specific, something like

`is_string` is a function that takes a `str | None` argument, and returns a boolean. You should make `is_string` be able to narrow the type of the argument based on its return value, i.e.
- When it returns True, narrow `value`'s type to `str`
- When it returns False, narrow `value`'s type to `None`

Reply:
In fact this challenge needs `typing.TypeGuard`, the input value could be `Any`, and it only narrows down the input type to `str` when the result is True and doesn't care otherwise.

Reply:
I see. how about this?

`is_string` is a function that takes an argument `value` of arbitrary type, and returns a boolean. You should make is_string be able to narrow the type of the argument based on its return value: when it's true, narrow `value`'s type to str

Reply:
Thanks, I updated the hint with more info.

Reply:
Thanks! Looks a lot better now